packages/stores/src/tx/tracer.ts

```diff
@@ -273,14 +334,19 @@ export class TxTracer {
           .toUpperCase()}'`,
       };
 
-      return new Promise<unknown>((resolve, reject) => {
-        this.txSubscribes.set(id, {
-          params,
-          resolver: resolve,
-          rejector: reject,
-        });
-
-        this.sendSubscribeTxRpc(id, params);
+      return new Promise<unknown>(async (resolve, reject) => {
+        try {
+          this.txSubscribes.set(id, {
+            params,
+            resolver: resolve,
+            rejector: reject,
+          });
+
+          this.sendSubscribeTxRpc(id, params);
+          resolve({ id });
+        } catch (e) {
+          reject(e);
+        }
```

Comment:
Note: this promise was not resolved properly. While I acknowledge that there are cleaner ways to address this by performing a bigger refactor to switch to async/await generally.

However, I opted in for a local change to limit the scope and risk around this critical but brittle part of the codebase.